src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -421,21 +473,35 @@ private KafkaRecordErrorReporter createKafkaRecordErrorReporter() {
                   errantRecordReporter.report(record, error).get();
                 } catch (InterruptedException | ExecutionException e) {
                   final String errMsg = "ERROR reporting records to ErrantRecordReporter";
-                  LOGGER.error(errMsg, e);
+                  this.DYNAMIC_LOGGER.error(errMsg, e);
                   throw new ConnectException(errMsg, e);
                 }
               };
         } else {
-          LOGGER.info("Errant record reporter is not configured.");
+          this.DYNAMIC_LOGGER.info("Errant record reporter is not configured.");
         }
       } catch (NoClassDefFoundError | NoSuchMethodError e) {
         // Will occur in Connect runtimes earlier than 2.6
-        LOGGER.info("Kafka versions prior to 2.6 do not support the errant record reporter.");
+        this.DYNAMIC_LOGGER.info(
+            "Kafka versions prior to 2.6 do not support the errant record reporter.");
       }
     }
     return result;
   }
 
+  private String getTaskLoggingTag() {
+    int countThreshold = 999;
+
+    if (totalTaskCreationCount > countThreshold) {
+      this.DYNAMIC_LOGGER.warn(
+          "More than {} tasks have been created. Resetting to 0", countThreshold);
+      totalTaskCreationCount = 0;
+    }
```

Comment:
why this is needed?

Reply:
This was created in the interest of reducing log size if the number got too large. I can remove this if we want